presto-main/src/main/java/com/facebook/presto/execution/TaskExecutor.java

```diff
@@ -605,7 +606,12 @@ public void run()
                         }
                     }
                     catch (Throwable t) {
-                        log.error(t, "Error processing %s", split.getInfo());
+                        if (t instanceof PrestoException) {
+                            log.error("Error processing %s: %s", split.getInfo(), t);
+                        }
+                        else {
```

Comment:
What types of Exceptions are expected here? Wouldn't it make sense to whitelist the expected the expected ones?